src/App.js

```diff
@@ -1,24 +1,79 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react';
 import './App.css';
+import foods from './foods.json';
+import 'bulma/css/bulma.css';
+import Foodbox from './components/Foodbox';
+import Search from './components/Search';
+
+// const foodsArr = foods;
+
+
+
 
 function App() {
+
+const [food, setFood] = useState(foods)
+const [name, setName] = useState("");
+const [image, setImage] = useState("");
+const [calories, setCalories] = useState("");
+const [search, setSearch] = useState("")
+
+
+
+  const handleNameChange = (e)=>{
+    console.log(e)
+    setName(e.target.value)
+  }
+  const handleImageChange = (e)=>{
+    setImage(e.target.value)
+  }
+  const handleCaloriesChange = (e)=>{
+    setCalories(e.target.value)
+  }
+
+  const handleSubmit = (e)=>{
+    e.preventDefault();
+    setFood((prevFood)=>{
+      const newFood ={
+        name: name,
+        image: image,
+        claories: calories
+      }
+      
+     const newFoodArr = [newFood, ...prevFood]
+     console.log(newFoodArr)
+     return newFoodArr;
+    })
+    return 
+  }
+
+  
+
+  const searchFood = (search)=>{
+    setSearch(search)
+    setFood((prevFood)=>{
+     const newArr = prevFood.filter((name)=>{
+      return name.name.includes(search)
+      })
+      const result = [...newArr];
+      return result;
+    })
+  }
```

Comment:
The way we structured our code now, we are sending 'filteredFood' variable to FoodBox, so this is the state we want to update when searching (filtering). 

However, we Always need to filter Original array --> foods. If you think about it, when you start typing smth in search input -> we already have filtered array, correct? If I then add another letter or remove one -> with your current code i would be filtering again on already  Filtered array.

So the idea is that we always read seach input value -> but filter original array 

```
  const searchFood = (search)=>{
    console.log(search)
    setFilteredFood(food.filter((name) => name.name.includes(search)))
}
```